src/catalog/schema.cpp

```diff
@@ -124,12 +125,74 @@ Schema *Schema::CopySchema(const Schema *schema) {
   return CopySchema(schema, set);
 }
 
-// Copy subset of columns in the given schema
+/*
+ * CopySchema() - Copies the schema into a new schema object with index_list
+ *                as indices to copy
+ *
+ * This function essentially does a "Gathering" operation, in a sense that it
+ * collects columns indexed by elemenys inside index_list in the given order
+ * and store them inside a newly created schema object
+ *
+ * If there are duplicates in index_list then the columns will be duplicated
+ * (i.e. no dup checking will be done)
+ *
+ * If the indices inside index_list >= the size of the column list then behavior
+ * is undefined (and is likely to crash)
+ *
+ * The returned schema is created by new operator, and the caller is responsible
+ * for destroying it.
+ */
 Schema *Schema::CopySchema(const Schema *schema,
-                           const std::vector<oid_t> &set) {
+                           const std::vector<oid_t> &index_list) {
+  std::vector<Column> column_list{};
+  
+  // Reserve some space to avoid multiple ma110��() calls
```

Comment:
No it's not. Since the validator forbids any occurrence of malloc() so I changed to ma1(one)1(one)0(zero)c(cent)